File: lib/java/src/main/java/io/github/brenbar/uapi/internal/GenerateRandomAny.java
package io.github.brenbar.uapi.internal;

import io.github.brenbar.uapi.RandomGenerator;

public class GenerateRandomAny {
    static Object generateRandomAny(RandomGenerator randomGenerator) {
        final var selectType = randomGenerator.nextIntWithCeiling(3);
        if (selectType == 0) {
            return randomGenerator.nextBoolean();
        } else if (selectType == 1) {
            return randomGenerator.nextInt();
        } else {
            return randomGenerator.nextString();
        }
    }
}

File: lib/go/internal/types/GenerateRandomAny.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package types

// GenerateRandomAny produces a pseudo-random primitive value.
func GenerateRandomAny(ctx *GenerateContext) any {
	if ctx == nil || ctx.RandomGenerator == nil {
		return nil
	}

	selectType := ctx.RandomGenerator.NextIntWithCeiling(3)
	switch selectType {
	case 0:
		return ctx.RandomGenerator.NextBoolean()
	case 1:
		return ctx.RandomGenerator.NextInt()
	default:
		return ctx.RandomGenerator.NextString()
	}
}
